packages/docusaurus-remark-plugin-npm2yarn/src/index.ts

```diff
@@ -6,45 +6,67 @@
  */
 
 import visit from 'unist-util-visit';
-// @ts-expect-error: this package provides CJS
 import npmToYarn from 'npm-to-yarn';
 import type {Code, Content, Literal} from 'mdast';
 import type {Plugin} from 'unified';
 import type {Node, Parent} from 'unist';
 
+type CustomConverter = [name: string, cb: (npmCode: string) => string];
+
 type PluginOptions = {
   sync?: boolean;
+  converters?: (CustomConverter | 'yarn' | 'pnpm')[];
 };
 
-// E.g. global install: 'npm i' -> 'yarn'
-const convertNpmToYarn = (npmCode: string) => npmToYarn(npmCode, 'yarn');
-
-const transformNode = (node: Code, isSync: boolean) => {
-  const groupIdProp = isSync ? ' groupId="npm2yarn"' : '';
-  const npmCode = node.value;
-  const yarnCode = convertNpmToYarn(node.value);
+function createTabItem(
+  code: string,
+  node: Code,
+  value: string,
+  label?: string,
+) {
   return [
     {
       type: 'jsx',
-      value: `<Tabs${groupIdProp}>\n<TabItem value="npm">`,
+      value: `<TabItem value="${value}"${label ? ` label="${label}"` : ''}>`,
     },
     {
       type: node.type,
       lang: node.lang,
-      value: npmCode,
+      value: code,
     },
     {
       type: 'jsx',
-      value: '</TabItem>\n<TabItem value="yarn" label="Yarn">',
+      value: '</TabItem>',
     },
+  ] as Content[];
+}
+
+const transformNode = (
+  node: Code,
+  isSync: boolean,
+  converters: (CustomConverter | 'yarn' | 'pnpm')[],
+) => {
+  const groupIdProp = isSync ? ' groupId="npm2yarn"' : '';
+  const npmCode = node.value;
+  return [
     {
-      type: node.type,
-      lang: node.lang,
-      value: yarnCode,
+      type: 'jsx',
+      value: `<Tabs${groupIdProp}>`,
     },
+    ...createTabItem(npmCode, node, 'npm'),
+    ...converters.flatMap((converter) =>
+      typeof converter === 'string'
+        ? createTabItem(
+            npmToYarn(npmCode, converter),
+            node,
+            converter,
+            converter === 'yarn' ? 'Yarn' : 'PnPm',
```

Comment:
Nit: the official repo https://github.com/pnpm/pnpm does not indicate that its normalized name is `PnPm`.

According to the benchmark label https://github.com/pnpm/pnpm#benchmark, it should be `pnpm`.